templates/concourse/base.lib.yml

```diff
@@ -32,6 +33,8 @@ plan:
 - #@ ytt_processor(env_name, processor, secrets)
 #@ if/end executor["type"] == "kapp":
 - #@ kapp_task(env_name, executor)
+#@ if/end executor["type"] == "terraform":
+- #@ terraform_task(env_name, executor["path"])
```

Comment:
instead of passing in `executor["path"]` can you just pass in `executor`. The `terraform_task` function should be the only place with knowledge of the schema of the `executor` field. This violates the law of Demeter.